post-processor/vsphere-template/step_create_folder.go

```diff
@@ -0,0 +1,72 @@
+package vsphere_template
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/multistep"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+)
+
+type StepCreateFolder struct {
+	Folder string
+}
+
+func (s *StepCreateFolder) Run(state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+	ctx := state.Get("context").(context.Context)
+	f := state.Get("finder").(*find.Finder)
+	d := state.Get("datacenter").(string)
+
+	if s.Folder != "" {
+		ui.Say("Creating or checking destination folders...")
+
+		if !strings.HasPrefix(s.Folder, "/") {
+			s.Folder = filepath.Join("/", s.Folder)
+		}
```

Comment:
What a mean is that canceling all because the folder path doesn't start with "/" could be a little restrictive, this check is only to build a correct path, for example: /vm/my/new/path where the folder is /my/new/path ( it could also be my/new/path). But if you still think that is necessary I could make the change.